File: 0056-merge-intervals/0056-merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        
        res = []
        
        for s,e in sorted(intervals):
            if not res or s>res[-1][1]:
                res.append([s,e])
            else:
                res[-1][1] = max(res[-1][1], e)
        
        return res

File: 0056-merge-intervals/0056-merge-intervals.java
class Solution {
    public int[][] merge(int[][] intervals) {
        Arrays.sort(intervals, (a,b) -> a[0]-b[0]);
        ArrayList<int[]> res = new ArrayList();
        for(int i=0;i<intervals.length;i++)
        {
            if(res.size()==0){
                res.add(intervals[i]);
                continue;
            }
            
            if(intervals[i][0]<=res.get(res.size()-1)[1])
                res.get(res.size()-1)[1] = Math.max(res.get(res.size()-1)[1],intervals[i][1]);
            else
                res.add(intervals[i]);
        }
        
        return res.toArray(new int[res.size()][]);
        
    }
}